docs/resource_sets.rst

```diff
@@ -91,3 +93,168 @@ Limitations
+TODO: "On a high level" -> remove?
+TODO: should I simplify and just talk about resources rather than generic nodes? Less accurate but also less abstract
+
+On a high level, to allow safe use of partial compiles, it is imperative that each service's refinements (through
+implementations) form a tree that can only intersect between service instances on shared nodes. The whole subtree below such a
+shared node should be considered shared and any resources in it must not be part of a resource set. All shared resources should
+be consistent between any two service instances that might create the object (see `Constraints and rules`_). All other nodes
+should generally be considered owned by the service and all their resources be part of the service's resource set. For more
+details on what it means to own a a child node in the tree (e.g. a resource) and how to ensure two service instance's trees can
+not intersect on owned nodes, see the `Ownership`_ subsection.
+
+Service instance uniqueness
+***************************
+With full compiles, indexes serve as the identity of a service instance. The compiler then validates that no conflicting
+service instances can exist. With partial compiles this validation is lost because only one service instance will be present
+in the model. However, it is still crucial that such conflicts do not exist. Put simply, we need to make sure that a partial
+compile succeeds only if a full compile would succeed as well. This subsection deals solely with the uniqueness of service
+instances. The `Ownership`_ subsection then deals with safe refinements into resources.
+
+To ensure service instance definitions are distinct, the model must make sure to do appropriate validation on the full set of
+definitions. When doing a partial compile, the model must verify that the service instance it is compiling for, has a different
+identity from any of the previously defined service instances. This can be achieved by externally checking against some sort of
+inventory that there are no matches for any set of input attributes that identify the instance.
+
+The current implementation of partial compiles does not provide any helpers for this verification. It is the responsibility of
+the model developer or the tool/extension that does the export to ensure that no two service instances can be created that are
+considered to have the same identity by the model.
+
+Ownership
+*********
+A resource can safely be considered owned by a service instance if it could never be created by another service instance. There
+are two main mechanisms that can be used to provide this guarantee. One is the use of indexes on appropriate locations, the
+other is the use of some external distributor of unique values (e.g. a plugin to generate a UUID or to allocate values in an
+inventory).
+
+In either case, the goal is to make sure that any object that is marked as owned by a service instance, is unique to that
```

Comment:
1. The index here is in fact not required, the condition below it is sufficient:
`we do so by making sure the object's identity is in fact completely derived from the identity of the service instance` 
The index helps verifying this behavior when doing a full compile, but it not required.

2. completely derived is not sufficient, if we say `the name is the first character of the service id`, the name is completely derived from the service id, but not unique

3. the service id is not the only valid option: if the service instance has other unique (combinations of fields), that are provided as unique (see previous section), they can also serve to make unique id's 


(point 2 and 3 can be considered too precise to bother)

Reply:
2. I did take that into account but I see how my wording is quite ambiguous in that respect. I'll pay special attention to this.
3. I don't think I ever mention "service id" directly (I've tried to explicitly avoid that). I mostly speak generically of "identity", which corresponds to what you're saying. I'll see if I can better define what I mean by that somewhere.

Reply:
2. I tried to improve this a bit.
3. Added a definition of what the document means when it says "identity".